IdentityCore/src/logger/MSIDMaskedLogParameter.m

```diff
@@ -50,7 +50,7 @@ - (instancetype)initWithParameterValue:(id)parameter
 
 - (NSString *)description
 {
-    if (![MSIDLogger sharedLogger].PiiLoggingEnabled)
+    if (!MSID_PII_LOGGING_ENABLED)
```

Comment:
Can you provide more context about this change and why MSIDLogger cannot be used directly?

Reply:
I'm now looking at a potential way to not need this change, but to explain what this relates to, the current way we're doing logging when building this as part of msalcpp is to override the MSID_LOG_xyz() macros in our builds so they invoke our own MSAILogging class instead of using MSIDLogger.

Reply:
I'm concerned about replacing code with macros just because we don't have a better way to coexist with CPP. I'd prefer to come up with a proper solution.

Reply:
I see 2 possible solutions:
- We can extract MSIDLogging interface and allow injecting it as a dependency.
- CPP team can subscribe to our logger callback and modify log output there.

Reply:
Updated so we can use the same MSID_LOG_xyz() macro definitions. We can override MSIDLogger's implementation in our builds to invoke our logging subsystem.